Dockerfile

```diff
@@ -26,16 +26,6 @@ COPY vast ./vast
 COPY CMakeLists.txt LICENSE VAST.spdx README.md VERSIONING.md \
      vast.yaml.example version.json ./
 
-# Resolve repository-internal symlinks.
-# TODO: We should try to get rid of these long-term, as Docker does not work
-# well with repository-internal symlinks. The the integration test symlinks we
-# can get rid of by copying the integration test directory to the build
-# directory when building VAST.
-RUN for f in contrib/vast-plugins/*/CMakeLists.txt; do \
-      name="$(basename "$(dirname "$f")")"; \
-      ln -sf "../contrib/vast-plugins/$name" "plugins/$name"; \
-    done
```

Comment:
This loop is not unnecessary. It allows `docker build .` to build an image with all proprietary plugins as you've checked them out locally.

Docker's COPY command does not copy symlinks.

The comment above, however, should probably be reworded.

Reply:
Experimentally it does work with symlinks, at least on my machine (docker 23.0.1)

That said, both versions are not ideal as they require you to modify your working tree in order to not build some of the plugins (preventing interactive rebases in the meantime). I think ideally we should have a script like `scripts/setup-closed-source-plugins.sh` that checks out the submodule and then symlinks (or copies) the plugins as needed.

Reply:
I used a different approach for development now, we should eventually address this in a separate PR.